src/types/closure.md

```diff
@@ -316,79 +316,85 @@ If a closure captures a field of a composite types such as structs, tuples, and
 ## Overall Capture analysis algorithm
 
 * Input:
-    * Analyzing the closure C yields a set of `(Mode, Place)` pairs that are accessed
+    * Analyzing the closure C yields a mapping of `Place -> Mode` that are accessed
     * Access mode is `ref`, `ref uniq`, `ref mut`, or `by-value` (ordered least to max)
+        * For a `Place` that is used in two different acess modes within the same closure, the mode reported from closure analysis is the maximum access mode.
+        * Note: `ByValue` use of a `Copy` type is seen as a `ref` access mode.
     * Closure mode is `ref` or `move`
 * Output:
-    * Minimal `(Mode, Place)` pairs that are actually captured
+    * Minimal `(Place, Mode)` pairs that are actually captured
 * Cleanup and truncation
     * Generate C' by mapping each (Mode, Place) in C:
-        * `(Mode1, Place1) = ref_opt(unsafe_check(copy_type(Mode, Place)))`
+        * `(Place1, Mode1) = ref_opt(unsafe_check(Place, Mode))`
         * if this is a ref closure:
-            * Add `ref_xform(Mode1, Place1)` to C'
+            * Add `ref_xform(Place1, Mode1)` to C'
         * else:
-            * Add `move_xform(Mode1, Place1)` to C'
+            * Add `move_xform(Place1, Mode1)` to C'
 * Minimization
     * Until no rules apply:
-        * For each two places (M1, P1), (M2, P2) where P1 is a prefix of P2:
+        * For each two places (P1, M1), (P2, M2) where P1 is a prefix of P2:
             * Remove both places from the set
-            * Add (max(M1, M2), P1) into the set
+            * Add (P1, max(M1, M2)) into the set
 * Helper functions:
-    * `copy_type(Mode, Place) -> (Mode, Place)`
-        * "By-value use of a copy type is a ref"
-        * If Mode = "by-value" and type(Place) is `Copy`:
-            * Return (ref, Place)
-        * Else
-            * Return (Mode, Place)
-    * `unsafe_check(Mode, Place) -> (Mode, Place)`
+    * `unsafe_check(Place, Mode) -> (Place, Mode)`
         * "Ensure unsafe accesses occur within the closure"
         * If Place contains a deref of a raw pointer:
             * Let Place1 = Place truncated just before the deref
-            * Return (Mode, Place1)
+            * Return (Place1, Mode)
         * If Mode is `ref *` and the place contains a field of a packed struct:
             * Let Place1 = Place truncated just before the field
-            * Return (Mode, Place1)
+            * Return (Place1, Mode)
         * Else
-            * Return (Mode, Place1)
-    * `move_xform(Mode, Place) -> (Mode, Place)` (For move closures)
+            * Return (Place, Mode)
```

Comment:
This I changed ~~Place1~~ Place, because i think it was a typo in the algorithm we wrote on Hackmd